service/history/api/create_workflow_util.go

```diff
@@ -125,18 +126,17 @@ func NewWorkflowWithSignal(
 		}
 	}
 
-	newWorkflowContext := workflow.NewContext(
-		shard.GetConfig(),
-		definition.NewWorkflowKey(
-			namespaceEntry.ID().String(),
-			workflowID,
-			runID,
-		),
-		shard.GetLogger(),
-		shard.GetThrottledLogger(),
-		shard.GetMetricsHandler(),
+	newWorkflowContext, releaseFn, err := cache.GetOrCreateWorkflowExecution(
```

Comment:
nit: can consistency checker be used here? I am fine either way I guess, I just think we don't have a consistent way of using consistency checker and workflow cache in api handlers.

I thought the intention for consistency checker is for API handler implementations to use it instead of cache directly. But I don't know why consistency checker also allows retrieving workflow cache. so maybe my understanding is wrong.

Reply:
We definitely don't have a consistent way to do that, I agree.

I just tried it, the problem with using the ConsistencyChecker is that it only has a `GetWorkflowLease` method; not a `GetOrCreateWorkflowLease` method. So it won't work since it tries to load the Workflow from the database.